app/views/events/_event_role_fields.html.erb

```diff
@@ -11,11 +11,11 @@
     <td><%= link_to_remove_fields image_tag("cross.png"), f, f.object.new_record? %></td>
   <% elsif f.object.member == current_member %>
     <td><%= f.object.role %></td>
-    <td><%= f.object.assigned_to use_display_name: true %></td>
+    <td><%= f.object.assigned_to(run_position_name_preference(current_member)) %></td>
     <td></td>
     <td><%= link_to_remove_fields image_tag("cross.png"), f %></td>
   <% else %>
     <td><%= f.object.role %></td>
-    <td><%= f.object.assigned_to use_display_name: true %></td>
+    <td><%= f.object.assigned_to(run_position_name_preference(current_member)) %></td>
```

Comment:
Are you sure that this doesn't error out for anonymous / not-logged-in users?

Reply:
Actually it does cause errors. The only place this is called is behind an auth guard (don't know if this is the ruby name) so in this case it should be impossible to error (before this PR it always showed the display name). I'll add a early return though to make this function open to be used in other places.